packages/wonder-blocks-tabs/src/hooks/use-tab-indicator.ts

```diff
@@ -112,31 +110,33 @@ export const useTabIndicator = (props: Props) => {
         };
     });
 
+    const positioningStyle = {
+        // Translate x position instead of setting the
+        // left position so layout doesn't need to be
+        // recalculated each time
+        transform: `translateX(${underlineStyle.left}px)`,
+        width: `${underlineStyle.width}px`,
+    };
+
     const indicatorProps: IndicatorProps = {
-        style: [
-            {
-                // Translate x position instead of setting the
-                // left position so layout doesn't need to be
-                // recalculated each time
-                transform: `translateX(${underlineStyle.left}px)`,
-                width: `${underlineStyle.width}px`,
-            },
-            styles.currentUnderline,
-            animated && styles.underlineTransition,
-            // We only show it once it is ready so that we don't show the
-            // indicator re-positioning itself after aphrodite styles are first
-            // loaded in the browser
-            !indicatorIsReady.current && {display: "none"},
-        ],
+        style: {
+            ...styles.currentUnderline,
+            ...positioningStyle,
+            ...(animated ? styles.underlineTransition : {}),
+            ...(!indicatorIsReady.current ? {display: "none"} : {}),
+        },
         role: "presentation",
     };
 
     return {indicatorProps, updateUnderlineStyle};
 };
 
-const styles = StyleSheet.create({
+// Styles for the tab indicator. We use the styles as inline styles instead of
+// aphrodite styles so that new classes aren't generated each time the tabs
+// resize.
```

Comment:
Switching to inline styles for the indicator which has dynamic styles also made the Chromatic snapshots more consistent for the StateSheet stories!

I think the issue was that the pseudo-states add on would add a class for the state, but it would get overridden when we updated the Aphrodite styles

Reply:
This makes perfect sense, thanks for this perf. improvement!